### PR TITLE
Change username to email

### PR DESCRIPTION
**Pull request by martijnjanssen**
All occurrences of username have been changed to email with the exception of some methods regarding logging in, in these cases the username keyword is critical.

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/products/controller/TicketRestController.java [hunk 1]
```diff
@@ -79,9 +79,9 @@ public TicketRestController(TicketService ticketService, OrderService orderServi
     @PreAuthorize("isAuthenticated()")
     @RequestMapping(value = "/tokens", method = RequestMethod.GET)
     public ResponseEntity<?> getTicketTokensOpenForTransfer(Authentication auth) {
-        UserDetails currentUser = (UserDetails) auth.getPrincipal();
+        User currentUser = (User) auth.getPrincipal();
         Collection<TicketTransferToken> tokens =
-                ticketService.getValidTicketTransferTokensByUser(currentUser.getUsername());
+                ticketService.getValidTicketTransferTokensByUser(currentUser.getEmail());
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/products/controller/TicketRestController.java]: Method name is wrong here (was, still is). Should either accept a `User`, or be changed to `getValidTicketTransferTokensByUserEmail()`

File: src/main/java/ch/wisv/areafiftylan/products/service/TicketService.java [hunk 2]
```diff
@@ -66,9 +66,9 @@ default Ticket requestTicketOfType(String type, List<String> options) {
      * Sets up the ticket for transfer
      *
      * @param ticketId     The ID of the ticket to be transferred
-     * @param goalUserName The name of the user which should receive the ticket
+     * @param goalEmail    The name of the user which should receive the ticket
      */
-    TicketTransferToken setupForTransfer(Long ticketId, String goalUserName);
+    TicketTransferToken setupForTransfer(Long ticketId, String goalEmail);
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/products/service/TicketService.java]: While we're changing names, doesn't `String receiverEmail` make more sense here?

File: src/main/java/ch/wisv/areafiftylan/products/service/TicketServiceImpl.java [hunk 3]
```diff
@@ -183,7 +183,7 @@ public TicketTransferToken setupForTransfer(Long ticketId, String goalUserName)
             throw new TicketAlreadyLinkedException();
         }
 
-        if (t.getOwner() != null && t.getOwner().getUsername().equals(goalUserName)) {
+        if (t.getOwner() != null && t.getOwner().getEmail().equals(goalEmail)) {
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/products/service/TicketServiceImpl.java]: Same here, `receiverEmail`

File: src/main/java/ch/wisv/areafiftylan/users/controller/CurrentUserRestController.java [hunk 4]
```diff
@@ -107,15 +106,14 @@
     @JsonView(View.Team.class)
     @RequestMapping(value = "/teams", method = RequestMethod.GET)
     public Collection<Team> getCurrentTeams(Authentication auth) {
-        UserDetails currentUser = (UserDetails) auth.getPrincipal();
-        return teamService.getTeamsByUsername(currentUser.getUsername());
+        User currentUser = (User) auth.getPrincipal();
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/users/controller/CurrentUserRestController.java]: I think you can plug in the User directly as a method argument. @MatthijsKok figured this out somewhere. Maybe we can even make it a Class field as every method in this Controller needs the User object.

File: src/main/java/ch/wisv/areafiftylan/users/controller/CurrentUserRestController.java [hunk 5]
```diff
@@ -107,15 +106,14 @@
...
+        return teamService.getTeamsByEmail(currentUser.getEmail());
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/users/controller/CurrentUserRestController.java]: Let's rename this method to `getTeamsByMemberEmail`

File: src/main/java/ch/wisv/areafiftylan/utils/db/migration/V20170507210556__rename_username_to_email.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package ch.wisv.areafiftylan.utils.db.migration;
+
+import org.flywaydb.core.api.migration.spring.SpringJdbcMigration;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+public class V20170507210556__rename_username_to_email implements SpringJdbcMigration {
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/utils/db/migration/V20170507210556__rename_username_to_email.java]: Aren't you forgetting a lot of other tables like Orders and ExpiredOrders?

File: src/test/java/ch/wisv/areafiftylan/UserRestIntegrationTest.java [hunk 7]
```diff
@@ -249,23 +249,7 @@ public void testCreateUserIsDisabled() {
     }
 
     @Test
-    public void createUserNoEmail() {
```
Comment (skamoen) [inline src/test/java/ch/wisv/areafiftylan/UserRestIntegrationTest.java]: Why is this removed?

File: src/main/java/ch/wisv/areafiftylan/utils/db/migration/V20170507210556__rename_username_to_email.java (hunk 6 above)
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/utils/db/migration/V20170507210556__rename_username_to_email.java]: Let me check, but as far as I remember these tables operate with the `Id` in the `Users` table and not the `Username` or `Email`.

File: src/test/java/ch/wisv/areafiftylan/UserRestIntegrationTest.java (hunk 7 above)
Comment (martijnjanssen) [inline src/test/java/ch/wisv/areafiftylan/UserRestIntegrationTest.java]: Because we already have a [`createUserMissingEmailField()`](https://github.com/AreaFiftyLAN/lancie-api/pull/406/files#diff-1a637b6c2fc39b5c44162281c7a849c2R252) test. It was left over from the move from `username` + `email` to only `email` as username.

File: src/main/java/ch/wisv/areafiftylan/users/controller/CurrentUserRestController.java (hunk 4 above)
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/users/controller/CurrentUserRestController.java]: We'll fix this in @MatthijsKok's PR #344.
